### ISSUE TITLE
vdirsyncer broken with current python-click

### ISSUE THREAD

**Issue by protist** [pimutils/vdirsyncer#887]
I just upgraded python-click (7.1.2 to 8.0.0), and vdirsyncer no longer works. If I revert python-click to 7.1.2, vdirsyncer works again. Possibly related to bypassed version checks at #831, but vdirsyncer should be updated to work with the latest python-click anyway.

* Your vdirsyncer version: 0.16.8
* If applicable, which server software (and which version) you're using: Nextcloud 21.0.1
* Your Python version: 3.9.5
* Your operating system: Arch Linux
* Your config file:
```
[general]
status_path = "~/.vdirsyncer/status/"

[pair nextcloud_contacts]
a = "nextcloud_contacts_local"
b = "nextcloud_contacts_remote"
collections = ["from a", "from b"]
conflict_resolution = ["command", "vimdiff"]

[storage nextcloud_contacts_local]
type = "filesystem"
path = "~/.contacts/"
fileext = ".vcf"

[storage nextcloud_contacts_remote]
type = "carddav"
url = "https://foo.com/nextcloud/remote.php/carddav/"
useragent = "Mozilla"
username = "protist"
password.fetch = ["command", "keyring", "get", "nextcloud", "protist"]
```
* Use `vdirsyncer -vdebug` for debug output. The output is sensitive, but
  please attach at least the last few lines before the error (if applicable),
  censored as necessary. This is almost always the most useful information.
```
$ vdirsyncer -vdebug sync
debug: Using 1 maximal workers.
error: Unknown error occurred: unmatched ')' (<string>, line 1)
error: Use `-vdebug` to see the full traceback.
debug:   File "/usr/lib/python3.9/site-packages/vdirsyncer/cli/__init__.py", line 30, in inner
debug:     f(*a, **kw)
debug:   File "/usr/lib/python3.9/site-packages/vdirsyncer/cli/__init__.py", line 145, in sync
debug:     with wq.join():
debug:   File "/usr/lib/python3.9/contextlib.py", line 117, in __enter__
debug:     return next(self.gen)
debug:   File "/usr/lib/python3.9/site-packages/vdirsyncer/cli/utils.py", line 364, in join
debug:     with ui_worker.patch_click():
debug:   File "/usr/lib/python3.9/contextlib.py", line 117, in __enter__
debug:     return next(self.gen)
debug:   File "/usr/lib/python3.9/site-packages/click_threading/__init__.py", line 144, in patch_click
debug:     with patch_ui_functions(wrapper):
debug:   File "/usr/lib/python3.9/contextlib.py", line 117, in __enter__
debug:     return next(self.gen)
debug:   File "/usr/lib/python3.9/site-packages/click_threading/monkey.py", line 50, in patch_ui_functions
debug:     stub_f = eval('lambda {s}: {n}._real_click_fn({a})'
```

**Comment by jturner314** [comment]
This issue affected me too. The problem is that `click-threading` is incompatible with the latest version of `click`. I've created click-contrib/click-threading#5 with a minimal fix. This seems to make `vdirsyncer` work properly on my machine.

**Comment by protist** [comment]
@jturner314 Thank you. That patch works perfectly for me too.

**Comment by raceface2nd** [comment]
I am not sure if this is related . . .

running vdirsyncer on command line is working. But as soon as I call vdirsyncer via crontab I get the following error. I piped the crontab output to a log file, which shows me this error.

```
Traceback (most recent call last):
  File "/usr/bin/vdirsyncer", line 11, in <module>
    load_entry_point('vdirsyncer==0.14.1', 'console_scripts', 'vdirsyncer')()
  File "/usr/lib/python3/dist-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 675, in main
    _verify_python3_env()
  File "/usr/lib/python3/dist-packages/click/_unicodefun.py", line 119, in _verify_python3_env
    'mitigation steps.' + extra)
RuntimeError: Click will abort further execution because Python 3 was configured to use ASCII as encoding for the environment.  Either run this under Python 2 or consult http://click.pocoo.org/python3/ for mitigation steps.

This system supports the C.UTF-8 locale which is recommended.
You might be able to resolve your issue by exporting the
following environment variables:

    export LC_ALL=C.UTF-8
    export LANG=C.UTF-8

```

**Comment by protist** [comment]
@raceface2nd I think that looks totally unrelated. Mine was also broken on the interactive CLI.

(I'd recommend starting a new thread, but did you try the suggestion? i.e. `export LC_ALL=C.UTF-8; export LANG=C.UTF-8; vdirsyncer…`)

**Comment by raceface2nd** [comment]
I wrote a small bash script now to call vdirsyncer. putting the exports before the call of vdirsyncer seems to do the trick

**Comment by dilyanpalauzov** [comment]
See also https://github.com/pallets/click/issues/1902 .

**Comment by WhyNotHugo** [comment]
This upstream issue is also a blocker: https://github.com/click-contrib/click-threading/pull/5

However, I believe that `click-threading` is unmaintained, so I might disable threading support if it's the only workaround.

**Comment by protist** [comment]
@WhyNotHugo That seems reasonable, but I'm not sure if you missed the reply to your linked issue? Looks like that issue was fixed?

**Comment by WhyNotHugo** [comment]
This is fixed in master with via #891
